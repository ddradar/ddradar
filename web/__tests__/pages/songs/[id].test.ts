import { testSongData } from '@ddradar/core/__tests__/data'
import Oruga from '@oruga-ui/oruga-next'
import { bulmaConfig } from '@oruga-ui/theme-bulma'
import { describe, expect, test, vi } from 'vitest'
import { ref } from 'vue'

import { useFetch, useRoute } from '#app'
import { mountAsync } from '~/__tests__/test-utils'
import Page from '~/pages/songs/[id].vue'

vi.mock('#app')

describe('Page /songs/[id]', () => {
  const params = { id: testSongData.id }

  test('{ isLoading: true } renders loading state', async () => {
    // Arrange
    /* eslint-disable @typescript-eslint/no-explicit-any */
    vi.mocked(useRoute).mockReturnValue({ params } as any)
    vi.mocked(useFetch).mockResolvedValue({
      pending: ref(true),
      data: ref(testSongData),
    } as any)
    /* eslint-enable @typescript-eslint/no-explicit-any */

    // Act
    const wrapper = await mountAsync(Page, {
      global: { plugins: [[Oruga, bulmaConfig]], stubs: { ChartInfo: true } },
    })

    // Assert
    expect(wrapper.element).toMatchSnapshot()
  })

  test('{ isLoading: false, song: <Data> } renders song info', async () => {
    // Arrange
    /* eslint-disable @typescript-eslint/no-explicit-any */
    vi.mocked(useRoute).mockReturnValue({ params } as any)
    vi.mocked(useFetch).mockResolvedValue({
      pending: ref(false),
      data: ref(testSongData),
    } as any)
    /* eslint-enable @typescript-eslint/no-explicit-any */

    // Act
    const wrapper = await mountAsync(Page, {
      global: { plugins: [[Oruga, bulmaConfig]], stubs: { ChartInfo: true } },
    })

    // Assert
    expect(wrapper.element).toMatchSnapshot()
  })
})
