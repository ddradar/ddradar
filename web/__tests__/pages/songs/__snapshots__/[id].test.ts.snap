// Vitest Snapshot v1

exports[`Page /songs/[id] > { isLoading: false, song: <Data> } renders song info 1`] = `
<section
  class="section"
>
  
  <h1
    class="title"
  >
    PARANOiA
  </h1>
  <h2
    class="subtitle"
  >
    180 / DDR 1st
  </h2>
  <h2
    class="subtitle"
  >
    BPM 180
  </h2>
  <div
    class="content columns is-multiline"
  >
    
    <chart-info-stub
      chart="[object Object]"
    />
    <chart-info-stub
      chart="[object Object]"
    />
    
  </div>
  <div
    class="content columns is-multiline"
  >
    
    
  </div>
  
</section>
`;

exports[`Page /songs/[id] > { isLoading: true } renders loading state 1`] = `
<section
  class="section"
>
  
  <transition-stub
    v-model:active="true"
  >
    <!--v-if-->
  </transition-stub>
  <h1
    class="title"
  >
    <div
      class="b-skeleton is-animated"
    >
      <div
        class="b-skeleton-item is-rounded"
      />
    </div>
  </h1>
  <h2
    class="subtitle"
  >
    <div
      class="b-skeleton is-animated"
    >
      <div
        class="b-skeleton-item is-rounded"
      />
    </div>
  </h2>
  <h2
    class="subtitle"
  >
    <div
      class="b-skeleton is-animated"
    >
      <div
        class="b-skeleton-item is-rounded"
      />
    </div>
  </h2>
  <div
    class="content columns is-multiline"
  >
    <div
      class="b-skeleton is-animated"
    >
      <div
        class="b-skeleton-item is-rounded"
      />
      <div
        class="b-skeleton-item is-rounded"
      />
      <div
        class="b-skeleton-item is-rounded"
      />
    </div>
  </div>
  
</section>
`;
